Added support for grabbing and putting data on S3.

shariqiqbal2810 opened the pull request:
* Added S3DataGrabber to nipype/interfaces/io.py
* Added S3DataSink to nipype/interfaces/io.py
* Added tests to nipype/interfaces/tests/test_io.py
    * Currently the S3 tests are using a bucket on our lab's account.
    * We would like to have a conversation about a more long-term testing solution.

satra commented:
for testing the grabber you can use the openfmri bucket (which is a public bucket). for the sink, i need to think of a better solution.

shariqiqbal2810 commented:
I changed the s3datagrabber test to use the openfmri data.  Unfortunately, these files are fairly large, so the tests are slowed (even though I'm only downloading 2).  Additionally, these modules require the use of boto3 (AWS Python API), and I was wondering what your solution would be to keep this optional. Currently I have this with my imports
```
try:
    import boto3
except:
    pass
```

shariqiqbal2810 commented:
FYI the only tests that are failing are due to a lack of boto3 on the testing system.  Tests all pass on my system.

chrisfilo commented:
We should skip those tests on systems without boto3 (since not all users
will use it), but also add it to circle.yml config file so it would be
installed on the test server.

On Fri, Sep 4, 2015 at 3:56 AM, Shariq Iqbal <notifications@github.com>
wrote:

> FYI the only tests that are failing are due to a lack of boto3 on the
> testing system. Tests all pass on my system.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/nipype/pull/1201#issuecomment-137557364>.
>

shariqiqbal2810 commented:
Ok done!  Any more thoughts on how to implement the S3DataSink test?

chrisfilo commented on nipype/interfaces/tests/test_io.py:
```diff
@@ -8,10 +8,17 @@
+    import boto3
+except:
+    noboto = True
+    pass
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Unnecessary pass statement](https://www.codacy.com/app/krzysztof-gorgolewski/nipy_nipype/file/935437085/issues/source?bid=2767207&fileBranchId=2841000#l20)

shariqiqbal2810 commented:
So I think the only remaining things to sort out are:
* Figuring out a more permanent testing solution for S3DataSink
* Getting AWS credentials for the testing server

chrisfilo commented:
What about using a fake S3 server? https://github.com/jubos/fake-s3

shariqiqbal2810 commented:
I modified the code to use a fake s3 server on the S3DataSink tests.  I added a 'gem install fakes3' line to the circle.yml file. Hopefully that will be enough to get everything working.
